aea/skills/base.py

```diff
@@ -61,34 +53,57 @@
 class SkillContext:
     """This class implements the context of a skill."""
 
-    def __init__(self, agent_context: AgentContext):
+    def __init__(
+        self,
+        agent_context: Optional[AgentContext] = None,
+        skill: Optional["Skill"] = None,
+    ):
         """
         Initialize a skill context.
-
-        :param agent_context: the agent's context
         """
-        self._agent_context = agent_context
+        self._agent_context = agent_context  # type: Optional[AgentContext]
         self._in_queue = Queue()  # type: Queue
-        self._skill = None  # type: Optional[Skill]
+        self._skill = skill  # type: Optional[Skill]
 
         self._is_active = True  # type: bool
         self._new_behaviours_queue = queue.Queue()  # type: Queue
         self._logger = None  # type: Optional[Logger]
 
+    @property
+    def logger(self) -> Logger:
+        """Get the logger."""
+        assert self._logger is not None, "Logger not set."
+        return self._logger
+
+    @logger.setter
+    def logger(self, logger_: Logger) -> None:
+        assert self._logger is None, "Logger already set."
+        self._logger = logger_
+
+    def _get_agent_context(self) -> AgentContext:
```

Comment:
why not a property?

Reply:
i see because it needs to be private..don't quite follow the change.

Reply:
to handle the case when `self._agent_context` is None. It avoids to repeat `assert self._get_agent_context is not None` in other properties.